moveit_core/collision_detection/src/world.cpp

```diff
@@ -131,14 +131,63 @@ void World::ensureUnique(ObjectPtr& obj)
     obj.reset(new Object(*obj));
 }
 
-bool World::hasObject(const std::string& id) const
+bool World::hasObject(const std::string& object_id) const
 {
-  return objects_.find(id) != objects_.end();
+  return objects_.find(object_id) != objects_.end();
 }
 
-bool World::moveShapeInObject(const std::string& id, const shapes::ShapeConstPtr& shape, const Eigen::Isometry3d& pose)
+bool World::knowsTransform(const std::string& object_id) const
 {
-  auto it = objects_.find(id);
+  // Check object names first
+  if (objects_.find(object_id) != objects_.end())
+    return true;
+  else  // Then objects' named frames
+  {
+    for (const auto& object : objects_)
+    {
+      if (object.second->named_frame_poses_.find(object_id) != object.second->named_frame_poses_.end())
+        return true;
+    }
+  }
+  return false;
+}
+
+const Eigen::Isometry3d& World::getTransform(const std::string& object_id) const
+{
+  auto it = objects_.find(object_id);
+  if (it != objects_.end())
+    return it->second->shape_poses_[0];
+  else  // Find within named frames
+  {
+    for (const auto& object : objects_)
+    {
+      if (object.second->named_frame_poses_.find(object_id) != object.second->named_frame_poses_.end())
+        return object.second->named_frame_poses_[object_id];
+    }
+  }
+  throw std::runtime_error("No transform found for object_id");
+}
+
+std::string World::getObjectOwningFrame(const std::string& frame_name) const
```

Comment:
Same comment here - i'd prefer that named frames within an object only make sense in the context of an object. So it wouldn't make sense to call `getObjectOwningFrame("handle_frame")` because i might have objects `"teapot1"` and `"teapot2"1 in the scene which both have a `"handle_frame"`

Reply:
See the mainline comment about namespacing